syntax = "proto3";

package serialization;

import "map_renderer.proto";

message RoutingSettings{
    int32 bus_wait_time = 1;
	double bus_velocity = 2;
}

message Coordinates{
	double lat = 1;
	double lng = 2;
}

message StopDistance{
	string stop_name_to = 1;
	int32 distance = 2;
}

message Stop{
	string stop_name = 1;
	Coordinates coordinates = 2;
	repeated StopDistance distance = 3;
}

message Bus{
	bool is_round = 1;
	string bus_name = 2;
	repeated string stop_names = 3;
}

message TransportDB{
	repeated Stop stops = 1;
	repeated Bus buses = 2;
    RenderSettings render_settings = 3;
    RoutingSettings routing_settings = 4;
}